src/number_theory/number_field.lean

```diff
@@ -234,7 +236,7 @@ begin
     exact (adjoin_root.lift_root hK).symm, },
 end
 
-variables (K A : Type*) [field K] [number_field K] [field A] [char_zero A] [is_alg_closed A] (x : K)
+variables (K A : Type*) [field K] [number_field K] [field A] [algebra ℚ A] [is_alg_closed A] (x : K)
```

Comment:
Is there a reason you made this change? I am not against it, but I think `[char_zero A]` is slightly more common (if you're still encountering the diamond, we should fix it rather than using a workaround).

Reply:
Yes, things go bad if I put ```char_zero A``` instead of ```algebra ℚ A```. I get a 
```invalid rewrite tactic, failed to synthesize type class instance for  char_zero A```
on line 292. I thought this instance would follow from the ```normed_algebra  ℚ A``` on line 259, but it doesn't in this case.

Reply:
I pushed a commit to use `char_zero`. Unfortunately, now we have to use `convert` in line 291, and this is because of che `char_zero` diamond. It should be solved soon hopefully.

Reply:
Ok. Thanks!

Reply:
I have split the result into two lemmas and the result is exactly of the same size. I also added some docstrings.